tests/unit/test_data_cleaning.py

```
@@ -0,0 +1,195 @@
+class TestDataFrameCleaning:
+    """Tests for DataFrame cleaning and reporting."""
+
+    def test_clean_dataframe_granular(self):
+        """Test clean_dataframe with mixed dirty data."""
+        test_df = pd.DataFrame({
+            'Dirty_Numeric': [">100", "1,234", "ERROR", "500"],
+            'Strictly_Text': ['A', 'B', 'C', 'D'],
+            'Mixed': ['10', '20', 'bad', 'worse']
+        })
+        
+        cleaned_df, report = clean_dataframe(test_df)
```

Ruff (0.14.13)

Unpacked variable `report` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)